batch/batch/server/server.py

```diff
@@ -451,16 +455,8 @@ def __init__(self, id, batch_id, attributes, callback, userdata, user, always_ru
             await self._delete_k8s_resources()
 
     async def delete(self):
-        children = [Job.from_record(record) for record in await db.jobs.get_children(self.id)]
```

Comment:
This is still wrong.  You can't set something to cancelled if it isn't ready.  I don't think we should have Job.delete, just call cancel.

Reply:
What about the log files?